### PR TITLE
Update contributor guide with updated scalar function howto

### LINKED ISSUE #9437: Update contributor guide with updated scalar function howto
### Is your feature request related to a problem or challenge?

Scalar functions have changed, update the contributor guide to reflect

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Omega359**
## Which issue does this PR close?

Closes #9437

### CODE REVIEW

File: docs/source/contributor-guide/index.md [hunk 1]
```diff
@@ -249,22 +249,23 @@ These are valuable for comparative evaluation against alternative Arrow implemen
...
-  - create a new module [here](../../../datafusion/physical-expr/src) for other functions
-- In [physical-expr/src](../../../datafusion/physical-expr/src/functions.rs), add:
-  - a new variant to `BuiltinScalarFunction`
-  - a new entry to `FromStr` with the name of the function as called by SQL
-  - a new line in `return_type` with the expected return type of the function, given an incoming type
-  - a new line in `signature` with the signature of the function (number and types of its arguments)
-  - a new line in `create_physical_expr`/`create_physical_fun` mapping the built-in to the implementation
-  - tests to the function.
+- Add the actual implementation of the function to a new module file within:
+  - [here](../../../datafusion/functions-array/src) for array functions
```
Comment (comphead) [inline docs/source/contributor-guide/index.md:253]: 👍

File: docs/source/contributor-guide/index.md [hunk 2]
```diff
@@ -249,22 +249,23 @@ These are valuable for comparative evaluation against alternative Arrow implemen
...
+  - [here](../../../datafusion/functions/src/datetime) for datetime functions
+  - [here](../../../datafusion/functions/src/encoding) for encoding functions
+  - [here](../../../datafusion/functions/src/math) for math functions
+  - [here](../../../datafusion/functions/src/regex) for regex functions
+  - [here](../../../datafusion/functions/src/string) for string functions
+  - create a new module [here](../../../datafusion/functions/src) for other functions. New function modules should likely have their own feature flag.
```
Comment (comphead) [inline docs/source/contributor-guide/index.md:259]:
```suggestion
  - create a new module [here](../../../datafusion/functions/src) for other functions.
```

This is slightly unclear as for me... Or we probably need to find better statement

Comment (Omega359) [reply docs/source/contributor-guide/index.md:259]: Ah, very true, I'll see if I can come up with something. I was trying to say that any new 'area' under functions, say datafusion/functions/src/vectors should be guarded by a feature flag.

**Review by alamb — APPROVED** [review]
Thanks @Omega359  and @comphead.

**Pull request merged**